lib/mongoid/slug.rb

```diff
@@ -90,8 +91,14 @@ def slug(*fields, &block)
 
         # Set indexes
         if slug_index && !embedded?
-          Mongoid::Slug::IndexBuilder.build_indexes(self, slug_scope_key, slug_by_model_type,
-                                                    options[:localize])
+          # Even if the slug_scope is nil, we need to proceed.
+          slug_scopes = slug_scope.nil? ? [slug_scope_key] : Array(slug_scope)
```

Comment:
The implementation of `slug_scope_key` iterates over slug scopes that can be `nil`, then returns `nil` in the case of `slug_scope` being nil.  

I suggest adding a `slug_scopes` method that does not call `slug_scope_key`:

```ruby
def slug_scopes
    slug_scope ? Array(slug_scope) : [nil]
end
```

Reply:
Got it.

Also I believe there's no need to change the way we invoke the index builder and we should keep it as it was in version 7.

```ruby
if slug_index && !embedded?
  Mongoid::Slug::IndexBuilder.build_indexes(self, slug_scope_key, slug_by_model_type,
                                            options[:localize])
end
```

But we've to modify the [index_builder.rb](https://github.com/mongoid/mongoid-slug/blob/6b6ad2f07f4772f5615dc8492c3ba870afb017c5/lib/mongoid/slug/index_builder.rb#L31) this way:

```diff
- fields[scope_key] = 1 if scope_key
+ Array(scope_key).each { |key| fields[key] = 1 }
```

Reply:
As long as all test pass ;)